### PR TITLE
Billing Address validation step

### PR DESCRIPTION
**Pull request by alex-deaconu**
Validate address with the Store server
Show validation errors if any
Receive confirmed address and update values entered

### CODE REVIEW

**Comment by alex-deaconu** [comment]
This PR took a different turn than expected. After discussions, we have removed the Address Confirmation step. Instead I opted for logging if the server returns any differences in the Address.

Also I'm planning to refactor a lot of this code for the Shipping Address implementation. Basically, make the Address code common for both. That's the main reason why I left out unit tests; I will recover the lost coverage in future PRs.

@Rise-Vision/apps please review. Thanks!

File: src/js/components/purchase-flow/dtv-billing-address.js [hunk 1]
```diff
@@ -15,6 +16,48 @@ angular.module("risevision.common.components.purchase-flow")
 
             return (field.$dirty || form.$submitted) && field.$invalid;
           };
+
+          var _addressesAreIdentical = function (src, result) {
+            var dest = {
+              street: result.line1,
+              unit: result.line2 && result.line2.length ? result.line2 : "",
```
Comment (fjvallarino) [inline src/js/components/purchase-flow/dtv-billing-address.js]: @alex-deaconu not sure what's the expected value for ```unit``` here

Comment (alex-deaconu) [reply src/js/components/purchase-flow/dtv-billing-address.js]: @fjvallarino `result` is just the response from the API, and this mapping for the comparison to the Address object is a straight copy from the Store (https://github.com/Rise-Vision/store/blob/master/web/script/ctr-edit-address-modal.js#L96).

Also, this is temporary code, trying to catch if there are any instances where the address returned is different from the one that was sent (and log in console). I will probably remove in the end.

Thanks.

Comment (fjvallarino) [reply src/js/components/purchase-flow/dtv-billing-address.js]: @alex-deaconu forget it

**Comment by fjvallarino** [comment]
@alex-deaconu looks good 👍

File: src/templates/components/purchase-flow/checkout-billing-address.html [hunk 2]
```diff
@@ -1,32 +1,38 @@
 <!-- Step 2 -->
-<div id="checkout-billing-address">
+<div id="checkout-billing-address" class="address-editor">
   <form id="form.billingAddressForm" role="form" class="u_margin-md-top" name="form.billingAddressForm" autocomplete="on" novalidate>
-    <div class="alert alert-danger" ng-show="form.billingAddressForm.$submitted && form.billingAddressForm.$invalid">Please complete the missing information below.</div>
+    <div class="alert alert-danger" ng-show="form.billingAddressForm.$submitted && form.billingAddressForm.$invalid">
+      Please complete the missing information below.
+    </div>
+    <div id="errorBox" ng-show="error" class="alert alert-danger" role="alert">
```
Comment (olegrise) [inline src/templates/components/purchase-flow/checkout-billing-address.html]: @alex-deaconu it doesn't look like you assign `error` in your code. I tried in the App and error message is not showing up.

File: src/templates/components/plans/plans-modal.html [hunk 3]
```diff
@@ -56,8 +56,8 @@ <h3>{{plan.proLicenseCount}}</h3>
           <!-- Trial expired notice -->
           <p class="small u_margin-sm-bottom text-danger" ng-show="isCurrentPlan(plan) && isTrialExpired(plan)" translate>common-header.plans.trial-expired</p>
 
-          <a id="subscribe-plan" ng-if="origin === 'Rise'" ng-show="getVisibleAction(plan) === 'subscribe'" class="btn btn-primary btn-block" target="_blank" href="https://store.risevision.com/product/{{plan.productId}}" translate>common-header.plans.subscribe</a>
-          <a id="subscribe-plan" ng-if="origin !== 'Rise'" ng-show="getVisibleAction(plan) === 'subscribe'" class="btn btn-primary btn-block" ng-click="purchaseFactory.showPurchaseModal(plan, isMonthly)" translate>common-header.plans.subscribe</a>
+          <a id="subscribe-plan" ng-if="origin === 'rise'" ng-show="getVisibleAction(plan) === 'subscribe'" class="btn btn-primary btn-block" target="_blank" href="https://store.risevision.com/product/{{plan.productId}}" translate>common-header.plans.subscribe</a>
```
[the remark this answers is not in the document]
Comment (alex-deaconu) [reply src/templates/components/plans/plans-modal.html]: @olegrise oops, I changed this locally so I could test with any Company as I don't know which one has origin=Chargebee. I will fix. Thanks.

File: src/templates/components/purchase-flow/checkout-billing-address.html (hunk 2 above)
Comment (alex-deaconu) [reply src/templates/components/purchase-flow/checkout-billing-address.html]: @olegrise will take a look, thanks.

**Comment by alex-deaconu** [comment]
@olegrise @fjvallarino there are too many issues with this PR. Main reason is the fact that all this code needs to be refactored for the shipping address implementation. Closing and will address all this in the next Shipping Address PR. Thanks.

**Comment by olegrise** [comment]
Oh, @alex-deaconu, you gave up 5 minutes before miracle happened. Just kidding. Sounds good :)

**Pull request closed without merging**